indykite/resource_trust_score_profile.go

```diff
@@ -129,9 +131,13 @@ func resourceTrustScoreProfileFlatten(
 
 	dimensions := make([]any, len(trustScoreProfile.GetDimensions()))
 	for i, dim := range trustScoreProfile.GetDimensions() {
+		floatNum, err := strconv.ParseFloat(fmt.Sprintf("%.3f", dim.GetWeight()), 64)
+		if err != nil {
+			d = append(d, buildPluginError("Error in converting weight"))
+		}
 		dimensions[i] = map[string]any{
 			trustScoreProfileName:   ReverseProtoEnumMap(TrustScoreDimensionNames)[dim.GetName()],
-			trustScoreProfileWeight: float32(dim.GetWeight()),
+			trustScoreProfileWeight: floatNum,
```

Comment:
Why converting float to string to convert back to float? Why you just cannot cast it?

Reply:
It is to remove the go round issue

Reply:
patch applied